server/lib/pbench/indexer.py

```diff
@@ -3157,15 +3157,15 @@ def hostnames_if_ip_from_sosreport(sos_file_name):
         hostname_s = hostname_f.split('.')[0]
     else:
         # both undefined
-        raise SosreportHostname("Hostname undefined (both short and long)")
+        return _dict_const([('message',"Hostname undefined (both short and long)")])
 
     if hostname_f == "localhost" and hostname_s != "localhost":
         hostname_f = hostname_s
         hostname_s = hostname_f.split('.')[0]
     elif hostname_f != "localhost" and hostname_s == "localhost":
         hostname_s = hostname_f.split('.')[0]
     elif hostname_f == "localhost" and hostname_s == "localhost":
-        raise SosreportHostname("The sosreport did not collect a hostname other than 'localhost'")
+        return _dict_const([('message',"The sosreport did not collect a hostname other than 'localhost'")])
```

Comment:
It's probably better to change the return type of the function to a tuple: (status, payload). The status tells us if everything went OK (in which case the payload is the host_md dict as happens now), or if an unrecoverable error occurred when processing this sosreport (in which case, the payload can be a description of the problem). But in either case, the payload is going to be sent to ES and indexed.

Reply:
ok